### PR TITLE
[doc] Add the workflow of the Auto-Pytorch

### PR DESCRIPTION
**Pull request by nabenabe0928**
Since the current repository and website does not have the detailed explanation how the Auto-Pytorch works, I added the descriptions of the workflow.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,6 +9,25 @@ We plan to enable image data and time-series data.
...
 Find the documentation [here](https://automl.github.io/Auto-PyTorch/development)
 
+## Workflow
+
+The rough description of the workflow of Auto-Pytorch is drawn in the following figure.
+
+<img src="figs/apt_workflow.png" width="500">
+
+In the figure, **Data** is provided by user and
+**Portfolio** is a predefined pool of machine learning algorithms, e.g. LightGBM and support vector machine, to solve either
```
Comment (ravinkohli) [inline README.md:19]: Portfolio is a greedy portfolio built based on the paper `AutoPyTorch Tabular`. They use the approach, PoSH-Auto-Sklearn to utilize portfolios in a meta-feature free approach

File: README.md [hunk 2]
```diff
@@ -9,6 +9,25 @@ We plan to enable image data and time-series data.
...
+regression or classification task on the provided dataset.
+Then API starts the following procedures:
+1. **Validate input data**: Process each data type, e.g. encoding categorical data, so that Auto-Pytorch can handled
+2. **Create dataset**: Create a dataset that can be handled in this API including cross validation splits
```
Comment (ravinkohli) [inline README.md:23]:
```suggestion
2. **Create dataset**: Create a dataset that can be handled in this API with a choice of cross validation or holdout splits
```

File: README.md [hunk 3]
```diff
@@ -9,6 +9,25 @@ We plan to enable image data and time-series data.
...
+3. **Evaluate baselines**: Train each algorithm in the predefined pool with a fixed hyperparameter configuration and dummy model from `sklearn.dummy` that represents the worst possible performance
```
Comment (ravinkohli) [inline README.md:24]: you can mention "Baselines is a predefined pool of machine learning algorithms, e.g. LightGBM and support vector machine, to solve either regression or classification task on the provided dataset" here.

File: README.md [hunk 4]
```diff
@@ -9,6 +9,25 @@ We plan to enable image data and time-series data.
...
+4. **Search by [SMAC](https://github.com/automl/SMAC3)**:\
+    a. Determine budget and cut-off rules by [Hyperband](https://jmlr.org/papers/volume18/16-558/16-558.pdf)\
+    b. Sample a target algorithm and the corresponding hyperparameter configuration by SMAC\
```
Comment (ravinkohli) [inline README.md:27]: what do you mean by sample a target algorithm? Do you mean the Algorithm Selection? 
```suggestion
    b. Sample a pipeline hyperparameter configuration (which specifies the choice of components in each step and their corresponding hyperparameters) by SMAC\
```

**Review by ravinkohli — CHANGES_REQUESTED** [review]
I have requested a few changes, otherwise I think it can be merged. I also feel we should include this information in the documentation,  do you have an idea where it can fit?

**Comment by ravinkohli** [comment]
> I have requested a few changes, otherwise I think it can be merged. I also feel we should include this information in the documentation, do you have an idea where it can fit?

Maybe in the developer documentation? https://automl.github.io/Auto-PyTorch/development/dev.html

**Comment by nabenabe0928** [comment]
> > I have requested a few changes, otherwise I think it can be merged. I also feel we should include this information in the documentation, do you have an idea where it can fit?
> 
> Maybe in the developer documentation? https://automl.github.io/Auto-PyTorch/development/dev.html

It seems the dev page is the only choice among all the pages.

**Pull request merged**